backend/editor/entries.py

```diff
@@ -5,270 +5,284 @@
+        return result
 
-def get_all_nodes(label):
-    """
-    Helper function used for getting all nodes with/without given label
-    """
-    qualifier = f":{label}" if label else ""
-    query = f"""
-        MATCH (n{qualifier}) RETURN n
-    """
-    result = get_current_transaction().run(query)
-    return result
+    def get_nodes(self, label, entry):
+        """
+        Helper function used for getting the node with given id and label
+        """
+        query = f"""
+            MATCH (n:{self.multi_label}:{label}) WHERE n.id = $id 
+            RETURN n
+        """
+        result = get_current_transaction().run(query, {"id": entry})
+        return result
 
-def get_nodes(label, entry):
-    """
-    Helper function used for getting the node with given id and label
-    """
-    query = f"""
-        MATCH (n:{label}) WHERE n.id = $id 
-        RETURN n
-    """
-    result = get_current_transaction().run(query, {"id": entry})
-    return result
+    def get_parents(self, entry):
+        """
+        Helper function used for getting node parents with given id
+        """
+        query = f"""
+            MATCH (child_node:{self.multi_label}:ENTRY)-[r:is_child_of]->(parent) WHERE child_node.id = $id 
+            RETURN parent.id
+        """
+        result = get_current_transaction().run(query, {"id": entry})
+        return result
 
-def get_parents(entry):
-    """
-    Helper function used for getting node parents with given id
-    """
-    query = f"""
-        MATCH (child_node:ENTRY)-[r:is_child_of]->(parent) WHERE child_node.id = $id 
-        RETURN parent.id
-    """
-    result = get_current_transaction().run(query, {"id": entry})
-    return result
+    def get_children(self, entry):
+        """
+        Helper function used for getting node children with given id
+        """
+        query = f"""
+            MATCH (child)-[r:is_child_of]->(parent_node:{self.multi_label}:ENTRY) WHERE parent_node.id = $id 
+            RETURN child.id
+        """
+        result = get_current_transaction().run(query, {"id": entry})
+        return result
 
-def get_children(entry):
-    """
-    Helper function used for getting node children with given id
-    """
-    query = f"""
-        MATCH (child)-[r:is_child_of]->(parent_node:ENTRY) WHERE parent_node.id = $id 
-        RETURN child.id
-    """
-    result = get_current_transaction().run(query, {"id": entry})
-    return result
+    def update_nodes(self, label, entry, new_node_keys):
+        """
+        Helper function used for updation of node with given id and label
+        """
+        # Sanity check keys
+        for key in new_node_keys.keys():
+            if not re.match(r"^\w+$", key) or key == "id":
+                raise ValueError("Invalid key: %s", key)
+        
+        # Get current node information and deleted keys
+        curr_node = self.get_nodes(label, entry, self.multi_label).data()[0]['n']

```

Comment:
I think we don't need the multi_label parameter:
```suggestion
        curr_node = self.get_nodes(label, entry).data()[0]['n']

```